python/mlc_llm/nn/kv_cache.py

```diff
@@ -976,12 +976,14 @@ def batch_decode_paged_kv(
                                 t0 = T.alloc_buffer((1,), "float32", scope="local")
 
                                 S_local = T.alloc_buffer((bdy * tile_size_per_bdx), "float32", scope="local")
-                                K_local = T.alloc_buffer((VEC_SIZE,), qkv_dtype, scope="local")
+                                K_local = T.alloc_buffer((VEC_SIZE,), "float32", scope="local")
```

Comment:
is this intentionally?

Reply:
This is to avoid extra copy from K_smem to K_local, now we are directly loading the Ksmem in compute statement itself and it helps to utilization vectorization and unroll reduction

It is make sense to change the naming convention for this K_local to QK_local. (Updated the commit)